src/Nri/Ui/Carousel/V2.elm

```diff
@@ -0,0 +1,344 @@
+    , previousButton : { name : String, icon : Svg, attributes : List (ClickableSvg.Attribute msg) }
+    , nextButton : { name : String, icon : Svg, attributes : List (ClickableSvg.Attribute msg) }
+    , name : String
+    , visibleLabelId : Maybe String
+    , role : Role
+    , announceAndSelect : { select : id, announce : String } -> msg
+    }
+    ->
+        { viewPreviousButton : Html msg
+        , viewNextButton : Html msg
+        , slides : Html msg
+        , containerAttributes : List (Attribute msg)
+        }
+viewWithPreviousAndNextControls config =
+    let
+        { viewPreviousButton, viewNextButton } =
+            case findPreviousAndNextSlides .id config of
+                Nothing ->
+                    { viewPreviousButton = text "", viewNextButton = text "" }
+
+                Just { previousSlide, nextSlide } ->
+                    { viewPreviousButton =
+                        viewSlideChangeButton
+                            { name = config.previousButton.name
+                            , icon = config.previousButton.icon
+                            , attributes = config.previousButton.attributes
+                            , targetSlideId = previousSlide.id
+                            , targetSlideLabel = previousSlide.name
+                            , carouselLabel = config.name
+                            , announceAndSelect = config.announceAndSelect
+                            }
+                    , viewNextButton =
+                        viewSlideChangeButton
+                            { name = config.nextButton.name
+                            , icon = config.nextButton.icon
+                            , attributes = config.nextButton.attributes
+                            , targetSlideId = nextSlide.id
+                            , targetSlideLabel = nextSlide.name
+                            , carouselLabel = config.name
+                            , announceAndSelect = config.announceAndSelect
+                            }
+                    }
+    in
+    { viewPreviousButton = viewPreviousButton
+    , viewNextButton = viewNextButton
+    , slides =
+        Html.div [ Accessibility.Styled.Key.tabbable False ]
+            (List.map
+                (\slide ->
+                    Html.div
+                        [ Role.group
+                        , Aria.roleDescription "slide"
+                        , id slide.idString
+                        , labelAttribute slide
+
+                        -- use as attribute for testing
+                        , if config.selected == slide.id then
+                            Attrs.style "display" "block"
+
+                          else
+                            Attrs.style "display" "none"
+                        ]
+                        [ slide.slideHtml ]
+                )
+                config.slides
+            )
+    , containerAttributes =
+        [ Attrs.attribute "role" (roleToString config.role)
```

Comment:
🎨 picky, so feel free to ignore, but maybe `roleToString` could change type and name to be `role -> Attribute msg` and use `Role.group` and `Role.region`

Reply:
Sure, just did it, the unique thing is that `Role.region` doesn't exist in this case